File: Python/Sorting/CountingSort.py
def countingSort(lst):
    k = max(lst)
    counts = [0] * (k + 1)
    for element in lst: #count elements
        counts[element] += 1
        
    index = 0
    for i, count in enumerate(counts):
        counts[i] = index
        index += count
        
    result = [0] * len(lst)
    for element in lst:
        idx = counts[element]
        result[idx] = element
        counts[element] += 1
        
    return result

File: Applications/CSharp/Sorting/CountingSort.cs
int[] CountingSort(int[] lst)
{
    int maximum = lst.Max();
    int[] counts = new int[maximum + 1];
    for (int i = 0; i < lst.Length; i++)
    {
        var element = lst[i];
        counts[element] += 1;
    }

    int idx = 0;
    for (int i = 0; i < counts.Length; i++)
    {
        var count = counts[i];
        counts[i] = idx;
        idx += count;
    }

    var result = new int[lst.Length];
    for (int i = 0; i < result.Length; i++)
    {
        var element = lst[i];
        result[counts[element]] = lst[i];
        counts[element] += 1;
    }

    return result;
}